src/middlewared/middlewared/plugins/virt/attachments.py

```diff
@@ -56,36 +61,137 @@ async def query(self, path, enabled, options=None):
-        ):
-            # If there are no attachments or if any attachment we have the virt pool
-            # being removed we should not do anything here
+        storage_pools = [p for p in virt_config['storage_pools'] if p != attachment['id']]
+        if attachment['incus_pool_change'] and attachment['id'] == virt_config['pool']:
+            # We are exporting main virt pool and at this point we should just unset
+            # the pool
+            await (await self.middleware.call('virt.global.update', {
+                'pool': None,
+                'storage_pools': storage_pools,
+            })).wait(raise_error=True)
             return
 
-        disks_to_remove = [i for i in filter(lambda i: i.get('disk_devices'), attachments)]
+        disks_to_remove = [i for i in filter(lambda i: i.get('disk_devices'), attachment['instances'])]
         for instance_data in disks_to_remove:
             for to_remove_disk in instance_data['disk_devices']:
                 await self.middleware.call('virt.instance.device_delete', instance_data['name'], to_remove_disk)
 
+        if attachment['incus_pool_change']:
+            # This means one of the storage pool is being exported
+            new_config = {
+                'pool': None,
+                'storage_pools': storage_pools,
+            }
+            await (await self.middleware.call('virt.global.update', new_config)).wait(raise_error=True)
+            await (await self.middleware.call(
+                'virt.global.update', {'pool': virt_config['pool']}
+            )).wait(raise_error=True)
+
     async def toggle(self, attachments, enabled):
+        await getattr(self, 'start' if enabled else 'stop')(attachments)
+
+    async def start(self, attachments):
+        if not attachments:
+            return
+
+        attachment = attachments[0]
+        if attachment['incus_pool_change']:
+            try:
+                await (await self.middleware.call('virt.global.setup')).wait(raise_error=True)
+            except Exception:
+                self.middleware.logger.error('Failed to start incus')
+                # No need to attempt to toggle instances, it won't happen either ways because none could be
+                # queried to be started as incus wasn't even running but better safe than sorry
+                return
+
+        await self.start_instances(attachment['instances'])
+
+    async def stop(self, attachments):
+        if not attachments:
+            return
+
+        attachment = attachments[0]
+        if attachment['incus_pool_change']:
+            # Stopping incus service does not stop the instances
+            # So let's make sure to stop them separately
+            await self.stop_running_instances()
+            try:
+                await self.middleware.call('service.stop', self.service)
+            except Exception as e:
+                self.middleware.logger.error('Failed to stop incus: %s', e)
```

Comment:
exc_info=True is probably better. If we do that, then the syslog logger will structure it better.